towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+
+        """
+        host = self.get_host_by_id(host_id)
+        return host.ansible_facts()
+
+    def get_all_inventories(self):
+        """
+        Get all the inventories of the ansible tower.
+
+        Returns:
+            list: list of inventories as EntityManager object.
+        """
+        inventories = [item for item in self.inventories]
+        return inventories
+
+    def get_all_hosts(self):
+        """Get all the hosts of the given ansible tower.
+
+        Returns:
+            list: list of all the hosts as EntityManager object.
+
+        """
+        hosts = [item for item in self.hosts]
+        return hosts
+
+    def get_all_credentials(self):
+        """Get all the credentials of the ansible tower.
+
+        Returns:
+            list: the list of credentials as EntityManager object.
+
+        """
+        credentials = [item for item in self.credentials]
+        return credentials
+
+    def get_hosts_by_inventory_id(self, inventory_id):
+        """Get filtered list of hosts for a given inventory id.
+
+        Args:
+             inventory_id: the given inventory id.
+
+        Returns:
+            list: the list of filtered host as EntityManger object.
+
+        """
+        hosts = [item for item in self.hosts if item.inventory.id == inventory_id]
+        return hosts
+
+    def get_all_jobs(self):
+        """Get all the jobs of the ansible tower.
+
+        Returns:
+            list: list of all the jobs as EntityManager object
+
+        """
+        jobs = [item for item in self.jobs]
+        return jobs
+
+    def get_jobs_by_name(self, given_job_name):
+        """Get filtered list of jobs for a given name.
+
+        Args:
+            given_job_name: the given job name.
+
+        Returns:
+             list: the filtered list of jobs.
+
+        """
+        job_list = [item for item in self.jobs if item.name == given_job_name]
+        return job_list
+
+    def get_project_updates(self):
+        """Get all the project updates for the ansible tower.
+
+        Returns:
+            list: list of all the project updates.
+
+        """
+        project_updates_url = '{api}/project_updates/'.format(api=self.api)
+        response = self.session.get(project_updates_url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json().get('results', [])
+
+    def get_project_updates_by_project_name(self, given_project_name):
+        """Get project update with the given project name.
+
+        Args:
+            given_project_name: the name of the project.
+
+        Returns:
+            dict: the project with all the information.
+
+        """
+        project_update_list = self.get_project_updates()
+        project_updates = [item for item in project_update_list if item['name'] == given_project_name]
```

Comment:
Please make key access safe by using .get method on it.

Reply:
refactored